prometheus_aws_guardduty_exporter/collector.py

```diff
@@ -22,53 +25,48 @@ def collect(self):
+            for severity, count in regionStats.items():
+                currentFindingsMetric.add_metric(value=count, labels=[region, severity])
 
         return [currentFindingsMetric, scrapeErrorsMetric]
 
-    def _collectMetricsByRegion(self, region, botoConfig):
-        client = boto3.client("guardduty", config=botoConfig, region_name=region)
+    def _collectMetricsByRegion(self, region):
+        client = boto3.client("guardduty", config=self.botoConfig, region_name=region)
         regionStats = {"low": 0, "medium": 0, "high": 0}
 
-        # List GuardDuty detectors
-        detectorIds = client.list_detectors()["DetectorIds"]
-
-        # Get statistics
-        for detectorId in detectorIds:
-            countBySeverity = client.get_findings_statistics(
-                DetectorId=detectorId,
-                FindingCriteria={"Criterion": {"service.archived": {"Eq": ["false"]}}},
-                FindingStatisticTypes=["COUNT_BY_SEVERITY"])["FindingStatistics"]["CountBySeverity"]
-
-            for severity, count in countBySeverity.items():
-                severity = float(severity)
-
-                # Group severity levels into low, medium and high according to this doc:
-                # https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_findings.html#guardduty_findings-severity
-                if severity < 4:
-                    regionStats["low"] += count
-                elif severity < 7:
-                    regionStats["medium"] += count
-                else:
-                    regionStats["high"] += count
-
-        return regionStats
+        try:
+            # List GuardDuty detectors
+            detectorIds = client.list_detectors()["DetectorIds"]
+
+            # Get statistics
+            for detectorId in detectorIds:
+                countBySeverity = client.get_findings_statistics(
+                    DetectorId=detectorId,
+                    FindingCriteria={"Criterion": {"service.archived": {"Eq": ["false"]}}},
+                    FindingStatisticTypes=["COUNT_BY_SEVERITY"])["FindingStatistics"]["CountBySeverity"]
+
+                for severity, count in countBySeverity.items():
+                    severity = float(severity)
+
+                    # Group severity levels into low, medium and high according to this doc:
+                    # https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_findings.html#guardduty_findings-severity
+                    if severity < 4:
+                        regionStats["low"] += count
+                    elif severity < 7:
+                        regionStats["medium"] += count
+                    else:
+                        regionStats["high"] += count
+        except Exception as error:
+            logging.getLogger().error(f"Unable to scrape GuardDuty statistics from {region} because of error: {str(error)}")
+
+            # Increase the errors count
+            self.scrapeErrors[region] += 1
```

Comment:
You're concurrently accessing this dictionary, because this method is executed in the thread. An option may be:
1. Return `regionStats` set to `False` on failure
2. In `collect()`, if `regionStats is False` you do `self.scrapeErrors[region] += 1`